### ISSUE TITLE
The problem of being unable to execute after modifying the port.

### ISSUE THREAD

**Issue by PlanetDance19** [ItzCrazyKns/Perplexica#515]
I still cannot run the application after changing the port.

I have OpenWebUI installed on my Mac, and it's occupying port 3000. So I modified the “config.toml” and “docker-compose.yaml” files in the perplexica folder, changing the default port 3001 to 4001. However, when I run "docker compose up -d", I get the error message: 'Error response from daemon: driver failed programming external connectivity on endpoint perplexica-perplexica-backend-1 (2ca26da26f8a95e1cae6311ea9551b836d9a4502c96e9a412322f996b123f12e): Bind for 0.0.0.0:4001 failed: port is already allocated'.

How can I solve this problem?

This is the revised document (original format could not be uploaded, so I converted it to txt format).
[config.txt](https://github.com/user-attachments/files/18088444/config.txt)
[docker-compose.txt](https://github.com/user-attachments/files/18088445/docker-compose.txt)

**Comment by ItzCrazyKns** [comment]
You made Perplexica front, backend and SearXNG on the same port, change port of one of them and update the addresses accordingly. Your issue should be fixed, make sure to not change the internal port of SearXNG from 8080 to anything else, the outside port can be configured.

**Comment by PlanetDance19** [comment]
> You made Perplexica front, backend and SearXNG on the same port, change port of one of them and update the addresses accordingly. Your issue should be fixed, make sure to not change the internal port of SearXNG from 8080 to anything else, the outside port can be configured.

Thank you for your reply. Although I've successfully deployed Perplexica, I cannot access it on the modified port 4000:4000. I noticed that the logs of "perplexica-perplexica-frontend-1" still show "Local:        http://localhost:3000".
![CleanShot 2024-12-11 at 13 58 23@2x](https://github.com/user-attachments/assets/84683bb1-1355-4466-8cac-9e430c572cd6)

The following is the file I modified. Is there anything I missed? Or are there other files that need to be changed?
[app.dockerfile.txt](https://github.com/user-attachments/files/18090310/app.dockerfile.txt)
[config.toml.txt](https://github.com/user-attachments/files/18090312/config.toml.txt)
[docker-compose.yaml.txt](https://github.com/user-attachments/files/18090314/docker-compose.yaml.txt)

**Comment by IMJONEZZ** [comment]
Same problem. I have OpenWebui occupying port 3000, so I changed the config.toml to put the Perplexica backend on port 3002, then changed the docker-compose.yaml accordingly, to put the backend on port 3002, the front-end on port 3001, and didn't modify the searxng instance. The service goes up, but when trying to find it on localhost:3001 or from another computer on the same network, it's just an "Unable to connect" error.

I noticed when running docker-compose logs -f these two lines:
`perplexica-backend-1 | No config path provided, using default path`
and 
`perplexica-frontend-1 | - Local: http://localhost:3000`

I'm not sure how to solve this, and any help would be appreciated.
